### PR TITLE
feat(discover): Display metrics baseline when conditions are met

### PR DESCRIPTION
**Pull request by shruthilayaj**
Introduces a MetricsBaseline component that wraps around
the results chart that queries for the timeseries data and
passes it to the `additionalSeries` prop in baseCharts.

### CODE REVIEW

File: static/app/views/eventsV2/metricsBaselineContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        } else {
+          const field = yAxis[0];
+          const prefixedName = `processed events: ${field}`;
+          const transformed = transformSeries(response, prefixedName, field);
+          additionalSeries.push(
+            LineSeries({
+              name: transformed.seriesName,
+              data: transformed.data.map(({name, value}) => [name, value]),
+              lineStyle: {type: 'dashed', width: 1, opacity: 0.5},
+              // itemStyle: {color: theme.gray200},
```
Comment (narsaynorath) [inline static/app/views/eventsV2/metricsBaselineContainer.tsx:150]:
```suggestion
```

File: static/app/views/eventsV2/metricsBaselineContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+  const disableProcessedBaselineToggle =
+    metricsCardinality.outcome?.forceTransactionsOnly ||
+    displayMode !== 'default' ||
+    !usesTransactionsDataset(eventView, yAxis);
+
+  const apiPayload = eventView.getEventsAPIPayload(location);
+  apiPayload.query = '';
+
+  const globalSelection = eventView.getPageFilters();
```
Comment (narsaynorath) [inline static/app/views/eventsV2/metricsBaselineContainer.tsx:64]: Nit: Should we rename`globalSelection` -> `pageFilters` here?

File: static/app/views/eventsV2/metricsBaselineContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    doEventsRequest(api, {
+      organization,
+      partial: true,
+      start,
+      end,
+      yAxis,
+      environment: globalSelection.environments,
+      period: globalSelection.datetime.period,
+      interval: eventView.interval,
```
Comment (narsaynorath) [inline static/app/views/eventsV2/metricsBaselineContainer.tsx:95]: Probably something to look into separately but I noticed that selecting 7d+ left some gaps when hovering with the tooltip. Adding a breakpoint here is giving me back `undefined` for `eventView.interval`


https://user-images.githubusercontent.com/22846452/191074979-4fa97764-2de5-4efa-8037-5a4a486fbed6.mov

**Review by narsaynorath — APPROVED** [review]
Just some small comments, but looks good to me. Tested it out and the metrics baseline looks pretty sweet!

File: static/app/views/eventsV2/metricsBaselineContainer.tsx (hunk 3 above)
Comment (shruthilayaj) [reply static/app/views/eventsV2/metricsBaselineContainer.tsx:95]: could perhaps be that intervals are mis-matched, will address in a follow up!

Comment (shruthilayaj) [reply static/app/views/eventsV2/metricsBaselineContainer.tsx:95]: Oh sorry just looked at the video and didn't read your comment, ok ty for checking

**Pull request merged**